### PR TITLE
Setup and connect bbb_api gem

### LINKED ISSUE #3220: Connect to BigBlueButton API gem
Variables should be configurable through the .env file

### PR DESCRIPTION
**Pull request by hadicheaito1**
closes #3220 

## Description

- Added bbb_api_ruby gem
- Added secret and url env variables
- Created BigBlueButtonApi service to allow the use of the bbb api
- Future methods to be made in the new service that access bbb api methods
- Added rspec basic test for checking that the instance was made
## Testing Steps

1. Pull code
2. Run 'bundle install'
3. Setup secret and url enviornment variables in .env
4. Add method in service that calls an api method such as (remember to remove after testing):
```
# Returns a list of all running meetings
  def all_running_meetings
    bbb_server.get_meetings
  end
```
5. Open rails console using 'rails c'
6. Create instance of BigBlueButtonApi service and call added method ```BigBlueButtonApi.new.get_meetings```
7. Confirm proper response is given indicating that the bbb api is connected and working correctly

### CODE REVIEW

**Comment by farhatahmad** [comment]
@hadicheaito1 Is this ready to be reviewed?

**Comment by hadicheaito1** [comment]
NOTE: SHA-256 implementation needs to kept in mind for later

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby '3.1.0'
 
 gem 'bcrypt', '~> 3.1.7'
+gem 'bigbluebutton-api-ruby', git: 'https://github.com/mconf/bigbluebutton-api-ruby.git', branch: 'master'
```
Comment (farhatahmad) [inline Gemfile:9]: We probably shouldn't pull from master

File: spec/services/big_blue_button_api_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+require 'bigbluebutton_api'
+
+describe BigBlueButtonApi, type: :service do
+  let(:bbb_server) { described_class.new }
+
+  describe 'Instance of BigBlueButtonApi created' do
+    it 'bbb_server is created and an instance of BigBlueBUttonApi' do
+      expect(bbb_server).to be_instance_of(described_class)
+    end
+  end
```
Comment (farhatahmad) [inline spec/services/big_blue_button_api_spec.rb:13]: This test doesn't actually test anything except Rails functionality.

Essentially what this test is doing is saying the BigBlueButtonApi.new returns an instance of BigBlueButtonApi

File: Gemfile (hunk 1 above)
Comment (farhatahmad) [reply Gemfile:9]: What's the latest version?

File: spec/services/big_blue_button_api_spec.rb (hunk 2 above)
Comment (farhatahmad) [reply spec/services/big_blue_button_api_spec.rb:13]: We could make the `bbb_server` method public and then test that it initializes an instance of BigBlueButton::BigBlueButtonApi with the right params passed in

File: spec/services/big_blue_button_api_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'bigbluebutton_api'
+
+describe BigBlueButtonApi, type: :service do
+  let(:bbb_server) { described_class.new }
```
Comment (farhatahmad) [inline spec/services/big_blue_button_api_spec.rb:7]: I would just go with `bbb` here

File: spec/services/big_blue_button_api_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  let(:bbb_api) { bbb_server.bbb_server }
```
Comment (farhatahmad) [inline spec/services/big_blue_button_api_spec.rb:8]: Does this need to be accessible for the entire service? if not, I would just move the logic down to the specific test itself (see below)

**Comment by farhatahmad** [comment]
@hadicheaito1 You included the `ruby-build` file in this PR

**Comment by farhatahmad** [comment]
@hadicheaito1 I refactored the tests a little. What do you think? 
```
before do
    ENV['BIGBLUEBUTTON_ENDPOINT'] = 'http://test.com/bigbluebutton/api'
    ENV['BIGBLUEBUTTON_SECRET'] = 'test'
  end

  let(:bbb_service) { described_class.new }

  describe 'Instance of BigBlueButtonApi being created' do
    it 'Created an instance of BigBlueButtonApi' do
      expect(BigBlueButton::BigBlueButtonApi).to receive(:new).with(ENV['BIGBLUEBUTTON_ENDPOINT'], ENV['BIGBLUEBUTTON_SECRET'], "1.8")
      bbb_service.bbb_server
    end

    it 'BigBlueButton client initialized once per request' do
      bbb_api = bbb_service.bbb_server
      bbb_api2 = bbb_service.bbb_server
      bbb_api3 = bbb_service.bbb_server

      expect(bbb_api).to eq(bbb_api2).and eq(bbb_api3)
    end
```

**Comment by hadicheaito1** [comment]
> @hadicheaito1 You included the `ruby-build` file in this PR

Yeah was an accident. I don't have access to removing it from the PR, and I can't undo the changes locally because that file is messed up locally (can't do anything with it). Based on searching online I think you have the rights to remove it from here?

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**